src/command/cmd_ac.c

```diff
@@ -4352,7 +4352,22 @@ _url_autocomplete(ProfWin* window, const char* const input, gboolean previous)
             return result;
         }
 
-        result = autocomplete_param_with_func(input, "/url save", wins_get_url, previous, window);
+        gboolean arg_result;
+        gchar** args = parse_args(input, 1, 8, &arg_result);
+        gboolean space_at_end = g_str_has_suffix(input, " ");
+        int num_args = g_strv_length(args);
+
+        if (arg_result) {
+            if ((num_args == 1 && space_at_end) || (num_args == 2 && !space_at_end)) {
+                result = autocomplete_param_with_func(input, "/url save", wins_get_url, previous, window);
+            } else if ((num_args == 2 && space_at_end) || (num_args == 3 && !space_at_end)) {
+                GString* beginning = g_string_new("/url save");
+                g_string_append_printf(beginning, " %s", args[1]);
+                result = cmd_ac_complete_filepath(input, beginning->str, previous);
+                g_string_free(beginning, TRUE);
```

Comment:
```suggestion
                gchar* cmd = g_strdup_printf("/url save %s", args[1]);
                result = cmd_ac_complete_filepath(input, cmd, previous);
                g_free(cmd);
```

`GString` is too fancy for that use case, also using `g_strdup_printf()` is IMO easier to read and a huge runtime overhead that will be avoided.